substrate/client/db/src/lib.rs

```diff
@@ -1693,27 +1680,29 @@ impl<Block: BlockT> Backend<Block> {
 							number,
 							hash,
 						)?;
-					}
-					if start > end {
-						transaction.remove(columns::META, meta_keys::BLOCK_GAP);
-						block_gap = None;
-						debug!(target: "db", "Removed block gap.");
-					} else {
-						block_gap = Some((start, end));
-						debug!(target: "db", "Update block gap. {:?}", block_gap);
-						transaction.set(
-							columns::META,
-							meta_keys::BLOCK_GAP,
-							&(start, end).encode(),
-						);
+						if start > end {
+							transaction.remove(columns::META, meta_keys::BLOCK_GAP);
+							block_gap = None;
+							debug!(target: "db", "Removed block gap.");
+						} else {
+							block_gap = Some((start, end));
+							debug!(target: "db", "Update block gap. {block_gap:?}");
+							transaction.set(
+								columns::META,
+								meta_keys::BLOCK_GAP,
+								&(start, end).encode(),
+							);
+						}
+						block_gap_updated = true;
```

Comment:
Not related nit: In below condition `number > best_num + One::one()` implies `number >One::one()`.
So the second condition can be removed

Reply:
Already addressed by https://github.com/paritytech/polkadot-sdk/pull/5343, feel free to ignore